src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_depth2img.py

```diff
@@ -474,7 +474,8 @@ def prepare_depth_map(self, image, depth_map, batch_size, do_classifier_free_gui
 
         # duplicate mask and masked_image_latents for each generation per prompt, using mps friendly method
         if depth_map.shape[0] < batch_size:
-            depth_map = depth_map.repeat(batch_size, 1, 1, 1)
+            repeat_by = batch_size // depth_map.shape[0]
+            depth_map = depth_map.repeat(repeat_by, 1, 1, 1)
```

Comment:
This will cause a mismatch in the batch dimension when using a batch size > 1 and a num_images_per_prompt > 1

It's always repeating by _total_ batch size even if the depth_map already is batched

Reply:
Good catch! Maybe having it as a comment would be helpful for us recollecting why this was needed.